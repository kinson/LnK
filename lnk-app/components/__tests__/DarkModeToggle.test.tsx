import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, queryByAttribute, render } from '@testing-library/react';
import DarkModeToggle from '../DarkModeToggle';

const getById = queryByAttribute.bind(null, 'id');

describe('DarkModeToggle component tests', () => {
  it('should render the DarkModeToggle component', () => {
    const component = renderer.create(<DarkModeToggle />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should toggle the darkMode when when the component is clicked', () => {
    const component = render(<DarkModeToggle />);

    const topLevelDiv = getById(
      component.container,
      'dark-mode-toggle-container'
    );

    expect(getById(component.container, 'moon-icon')).toBeTruthy();
    expect(getById(component.container, 'sun-icon')).toBeFalsy();

    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    fireEvent.click(topLevelDiv!);

    expect(getById(component.container, 'sun-icon')).toBeTruthy();
    expect(getById(component.container, 'moon-icon')).toBeFalsy();
  });
});
