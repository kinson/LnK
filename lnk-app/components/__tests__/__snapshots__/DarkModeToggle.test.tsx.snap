// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkModeToggle component tests should render the DarkModeToggle component 1`] = `
<button
  className="ml-auto mr-2 flex items-center cursor-pointer"
  id="dark-mode-toggle-container"
  onClick={[Function]}
  type="button"
>
  <div
    className="relative"
  >
    <input
      aria-label="Dark Mode Toggle"
      className="sr-only"
      defaultChecked={false}
      id="dark-mode-toggle"
      type="checkbox"
    />
    <div
      className="block bg-gray-600 w-14 h-8 rounded-full"
    />
    <div
      className="false absolute left-1 top-1 bg-white dark:bg-purple w-6 h-6 rounded-full flex items-center justify-center transition-all"
    >
      <svg
        className="h-4 text-black"
        fill="currentColor"
        id="sun-icon"
        stroke="currentColor"
        version="1.1"
        viewBox="0 0 170 170"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="Group-2"
          transform="translate(80.000000, 79.000000) rotate(-4.000000) translate(-80.000000, -79.000000) translate(1.000000, -0.000000)"
        >
          <circle
            cx="79"
            cy="79"
            id="Oval"
            r="22.5"
          />
          <g
            id="Group"
            stroke="currentColor"
            strokeLinecap="square"
            strokeWidth="5"
          >
            <line
              x1="111.085701"
              x2="139.247251"
              y1="51.5309469"
              y2="27.5243795"
            />
            <line
              x1="121.11157"
              x2="158"
              y1="82.2644628"
              y2="85.2024793"
            />
            <line
              transform="translate(118.472337, 125.166476) rotate(90.000000) translate(-118.472337, -125.166476) "
              x1="104.160729"
              x2="132.783944"
              y1="137.400592"
              y2="112.93236"
            />
            <line
              x1="75.7355372"
              x2="72.7975207"
              y1="121.11157"
              y2="158"
            />
            <line
              transform="translate(32.833524, 118.472337) rotate(180.000000) translate(-32.833524, -118.472337) "
              x1="18.7527491"
              x2="46.9142993"
              y1="130.475621"
              y2="106.469053"
            />
            <line
              x1="36.8884298"
              x2="0"
              y1="75.7355372"
              y2="72.7975207"
            />
            <line
              transform="translate(39.527663, 32.833524) rotate(270.000000) translate(-39.527663, -32.833524) "
              x1="25.2160557"
              x2="53.8392707"
              y1="45.0676403"
              y2="20.5994081"
            />
            <line
              x1="82.2644628"
              x2="85.2024793"
              y1="36.8884298"
              y2="0"
            />
          </g>
        </g>
      </svg>
    </div>
  </div>
</button>
`;
